docs/docs/adding-app-and-website-functionality.md

```diff
@@ -0,0 +1,78 @@
+
+1. Build and render static HTML, creating content and pages with data injected at build time
+2. Invoke `ReactDOM.hydrate()` method to pick up where the static HTML was left
+3. Transfer rendering to the React reconciler
+
+It's this last phase that bridges the gap between static sites and full-fledged applications. In this phase you can make calls for [dynamic data](/docs/client-data-fetching/), [authenticate users](/docs/building-a-site-with-authentication/), and perform all the app-like functionality you desire because the page is running a React application.
+
+## Common patterns for Gatsby apps
+
+There are different options for organizing how your pages are created and what they will be responsible for. These patterns can be combined and tweaked for specific use cases such as pulling in data at [build time](/docs/glossary#build) for great performance, or calling for data at [runtime](/docs/glossary#runtime) for a more dynamic experience.
+
+Because all Gatsby pages are hydrated into React, **any of the following patterns are capable of app-like behavior**. This section is to help explain some higher level patterns for thinking about Gatsby.
+
+### Static pages
+
+Static files are output by running `gatsby build` from exported components in your `src/pages` folder or from pages created using the [`createPage` API](/docs/node-apis/#createPages), as shown in this diagram:
+
+![Static Site diagram with pages created from Gatsby automatically and programmatically](./images/simple-static-site.png)
+
+The diagram illustrates the 2 main methods for creating pages in your site:
+
+1. Automatically through `src/pages`
+2. Programmatically with the `createPage` API
+
+_**Note**: plugins and themes can also implement the `createPage` API and create pages on your behalf_
+
+When you export a React component from a file in the `src/pages` directory (in this case `src/pages/home.js`) Gatsby will automatically create a static page. By looping through markdown files in your filesystem you can create pages for all blog posts programmatically. The docs have more informataion about [creating and modifying pages](/docs/creating-and-modifying-pages/).
+
+These created pages _could_ run JavaScript once React hydrates them, but they don't need to.
+
+### Hybrid app pages
+
+Your created pages can make calls to external services and APIs in order to allow more interactive and dynamic behavior. These are sometimes referred to as hybrid app pages because they share a combination of static features that Gatsby creates to help load your site quickly and performantly as well as traditional web app features.
+
+The following diagram shows a similar site to the one in the previous example, but with a subscription form added to the home page that makes `POST`'s to a database somewhere, as well as blog posts that `GET` recommendations from a database or other API.
+
+![Hybrid Site diagram with pages hitting other services with JavaScript](./images/simple-hybrid-site.png)
```

Comment:
<img width="597" alt="simple-hybrid-site" src="https://user-images.githubusercontent.com/21114044/68419328-3f0fbd00-0157-11ea-9cf5-b637460b3207.png">

Here's the update I just pushed, do you think that encapsulates the asynchronicity (is that a word?) okay?

Reply:
i think that helps, but I'm curious if having multiple terms would be misleading at all. What exact async tech is used depends on the imported / implemented library by the user, right?

Reply:
Right. You could hit the external service however, I figure they are all _somewhat_ synonymous and you could use any one of those methods or some other library that implements them for you. 

My rationale was if we say AJAX and someone has only heard of using the Fetch API this might help bridge the idea and see that it's not opinionated as to how you get data. Just leaving it as POST/GET kind of helps with that since those are the generic HTTP methods but I'm open to .

Reply:
Update: that seems fine after thinking about it further -- let's ship it!